### DISCUSSION TITLE
Better "Anonymous extension declarations" variant

### DISCUSSION THREAD

**Discussion by OJacot-Descombes** [dotnet/csharplang#8571] [LDM Notes]
In [Anonymous extension declarations](https://github.com/dotnet/csharplang/blob/main/meetings/working-groups/extensions/anonymous-extension-declarations.md#anonymous-extension-declarations) `extension(T){ ... }` parts are nested in a static class.

I suggest nesting `for(T){ ... }` parts in an `extension` instead:
``` c#
extension Enumerable
{
    // Extension members for IEnumerable
    for(IEnumerable)
    {
        // 'this' refers to underlying value
        public bool IsEmpty => !this.GetEnumerator().MoveNext();
    }

    // Extension members for IEnumerable<T>
    for<T>(IEnumerable<T>)
    {
        public IEnumerable<T> Where(Func<T, bool> predicate) { ... }
        public IEnumerable<TResult> Select<TResult>(Func<T, TResult> selector) { ... }
        public static IEnumerable<T> operator +(IEnumerable<T> e1, IEnumerable<T> e2) { ... }
    }
}
```

**Comment by HaloFour** [comment]
I don't think a separate discussion needs to be open for each variation on this syntax.  That's why the [design space discussion](https://github.com/dotnet/csharplang/issues/8548) exists.

This syntax flavor had already been suggested by the LDM: https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-10-14.md

I personally don't find it better.  It still runs into the same problems when trying to design for existing extension methods, in that the static container class may also contain non-extension helper methods.  Otherwise it just trades one keyword for another.
